### PR TITLE
Integrate AC permissions changes

### PR DESCRIPTION
**Pull request by Baron-Severin**
Waiting on https://github.com/mozilla-mobile/android-components/pull/5953

### Pull Request checklist

- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
No visual changes
- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsFragmentStore.kt [hunk 1]
```diff
@@ -351,7 +363,7 @@ sealed class WebsitePermission {
         override val isVisible: Boolean,
         override val isEnabled: Boolean,
         override val isBlockedByAndroid: Boolean,
-        val name: String = "Notification" // helps to resolve the overload resolution ambiguity for the copy() method
+        val name: String = "AutoplayAudible" // helps to resolve the overload resolution ambiguity for the copy() method
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsFragmentStore.kt]: I think this was by mistake?

File: app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsFragmentStore.kt [hunk 2]
```diff
@@ -390,6 +402,55 @@ sealed class WebsitePermission {
...
+
+    /**
+     * Contains all information about the *autoplay audible* permission.
+     */
+    data class AutoplayAudible(
+        override val status: String,
+        override val isVisible: Boolean,
+        override val isEnabled: Boolean,
+        override val isBlockedByAndroid: Boolean,
+        val name: String = "Location" // helps to resolve the overload resolution ambiguity for the copy() method
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsFragmentStore.kt]: nit:AutoplayAudible?

File: app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsManagePhoneFeatureFragment.kt [hunk 3]
```diff
@@ -102,14 +103,9 @@ class SitePermissionsManagePhoneFeatureFragment : Fragment() {
             append(recommendedSpannable)
             this
         }
-        val expectedAction = if (phoneFeature == PhoneFeature.AUTOPLAY) BLOCKED else ASK_TO_ALLOW
+        val expectedAction = if (phoneFeature == PhoneFeature.AUTOPLAY_AUDIBLE) BLOCKED else ASK_TO_ALLOW
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsManagePhoneFeatureFragment.kt]: @Baron-Severin should the `else` return `ALLOWED`  ?

Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsManagePhoneFeatureFragment.kt]: Spoke about this, the code is a little confusing but working as intended.

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsManagePhoneFeatureFragment.kt]: 👍

**Pull request merged**